jetty-servlet/src/main/java/org/eclipse/jetty/servlet/DefaultServlet.java

```diff
@@ -467,10 +467,31 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
             // Not found?
             if (content==null || !content.getResource().exists())
             {
-                if (included)
-                    throw new FileNotFoundException("!" + pathInContext);
-                response.sendError(HttpServletResponse.SC_NOT_FOUND);
-                return;
+                if (gzippable)
```

Comment:
I like that this only kicks in if the main resource is not found!  So that alleviates my performance concerns.  However I don't like that this code duplicates the setting of gzip headers etc.

I'd rather this code created a fake content object that returns the gz content from the getGzipContent() method, it can then fall through to all the normal handling below.   The fake content object needs to be good enough to allow any conditional headers to be answered so 206 responses can be sent.  It can probably achieve this by delegating the the gzip content for most methods.  Etags might be a little funny, but not a problem as there will be no etag for the normal content.